database/mysql/mysql_db.py

```diff
@@ -338,8 +340,19 @@ def main():
                     changed = db_create(cursor, db, encoding, collation)
                 except Exception, e:
                     module.fail_json(msg="error creating database: " + str(e))
+            module.exit_json(changed=changed, db=db)
+
+        elif state == "import":
+            if module.check_mode:
+                changed = True
+            else:
+                try:
+                    changed = db_create(cursor, db, encoding, collation)
+                except Exception, e:
```

Comment:
Looking at how `db_import `works, the `db_create` is probably not needed at all - calling `db_import` should do it all. That also means it'll cope better when `db` is `all`, and you won't create a db with the wrong collation or encoding values. This means that the import logic is the same whether or not the DB exists.